### PR TITLE
Get msg details

### PR DESCRIPTION
**Pull request by shohamroditimemphis**

### CODE REVIEW

File: ui_src/src/domain/stationOverview/hooks/reducer.js [hunk 1]
```diff
@@ -53,6 +53,11 @@ const Reducer = (stationState, action) => {
                 ...stationState,
                 selectedRowId: action.payload
             };
+        case 'SET_DLS_TYPE':
```
Comment (avrhamNeeman) [inline ui_src/src/domain/stationOverview/hooks/reducer.js:56]: You don't need this, you have isFailedSchemaMessage props

File: ui_src/src/domain/stationOverview/stationObservabilty/components/messageDetails/index.js [hunk 2]
```diff
@@ -55,9 +55,9 @@ const MessageDetails = ({ isDls, isFailedSchemaMessage = false }) => {
         try {
             const data = await httpRequest(
                 'GET',
-                `${ApiEndpoints.GET_MESSAGE_DETAILS}?station_name=${stationName}&is_poison_message=${isDls}&message_id=${
+                `${ApiEndpoints.GET_MESSAGE_DETAILS}?station_name=${stationName}&is_dls=${isDls}&message_id=${
                     isDls ? encodeURIComponent(selectedRow?.id) : null
-                }&message_seq=${selectedRow?.seq}`
+                }&message_seq=${selectedRow?.seq}&dls_type=${stationState.dlsType}`
```
Comment (avrhamNeeman) [inline ui_src/src/domain/stationOverview/stationObservabilty/components/messageDetails/index.js:60]: use with the isFailedSchemaMessage prop

File: ui_src/src/domain/stationOverview/stationObservabilty/messages/index.js [hunk 3]
```diff
@@ -63,7 +62,7 @@ const Messages = () => {
 
     const onCheckedAll = () => {
         setIsCheckAll(!isCheckAll);
-        subTabValue === 'Poison'
+        stationState?.dlsType === 'Poison'
```
Comment (avrhamNeeman) [inline ui_src/src/domain/stationOverview/stationObservabilty/messages/index.js:65]: Why use the state when you have the current reference?

File: server/memphis_handlers_dls_messages.go [hunk 4]
```diff
@@ -331,7 +331,17 @@ func getDlsMessageById(station models.Station, sn StationName, dlsMsgId string)
...
 	}
 
-	filterSubj := GetDlsSubject("*", sn.Intern(), dlsMsgId, ">")
+	dlsType = strings.ToLower(dlsType)
+	var filterSubj string
+	switch dlsType {
+	case "poison":
+		filterSubj = GetDlsSubject(dlsType, sn.Intern(), dlsMsgId, ">")
+	case "schema":
+		filterSubj = GetDlsSubject("schema", sn.Intern(), dlsMsgId, "")
```
Comment (idanasulinmemphis) [inline server/memphis_handlers_dls_messages.go:340]: Pass the dlsType also

File: server/memphis_handlers_stations.go [hunk 5]
```diff
@@ -1074,7 +1074,7 @@ func (sh StationsHandler) GetPoisonMessageJourney(c *gin.Context) {
 		return
 	}
 
-	poisonMessage, err := sh.GetDlsMessageJourneyDetails(body.MessageId)
+	poisonMessage, err := sh.GetDlsMessageJourneyDetails(body.MessageId, "Poison")
```
Comment (idanasulinmemphis) [inline server/memphis_handlers_stations.go:1077]: Poison with p instead of P

File: server/memphis_handlers_ws.go [hunk 6]
```diff
@@ -174,7 +174,7 @@ func memphisWSGetReqFillerFromSubj(s *Server, h *Handlers, subj string) (memphis
 			return nil, errors.New("invalid poison msg id")
 		}
 		return func() (any, error) {
-			return h.Stations.GetDlsMessageJourneyDetails(poisonMsgId)
+			return h.Stations.GetDlsMessageJourneyDetails(poisonMsgId, "Poison")
```
Comment (idanasulinmemphis) [inline server/memphis_handlers_ws.go:177]: p instead of P

**Review by idanasulinmemphis — APPROVED** [review]

**Pull request merged**